cloudinit/handlers/shell_script_by_frequency.py

```diff
@@ -0,0 +1,84 @@
+# part-handler
+# vi: syntax=python ts=4
+
+import os
+from cloudinit import log
+from cloudinit import util
+from cloudinit.handlers import Handler
+from cloudinit.settings import PER_ALWAYS, PER_INSTANCE, PER_ONCE
+LOG = log.getLogger(__name__)
+
+pathMap = {
+    PER_ALWAYS: 'per-boot',
+    PER_INSTANCE: 'per-instance',
+    PER_ONCE: 'per-once'
+}
+
+
+def get_script_folder_by_frequency(freq, scripts_dir):
+    freqPath = pathMap[freq]
+    folder = os.path.join(scripts_dir, freqPath)
+    return folder
+
+
+def write_script_by_frequency(script_path, payload, frequency, scripts_dir):
+    filename = os.path.basename(script_path)
+    filename = util.clean_filename(filename)
+    folder = get_script_folder_by_frequency(frequency, scripts_dir)
+    path = os.path.join(folder, filename)
+    payload = util.dos2unix(payload)
+    util.write_file(path, payload, 0o700)
+
+
+# per-boot
+class ShellScriptPerBootPartHandler(Handler):
+    def __init__(self, paths, **_kwargs):
+        Handler.__init__(self, PER_ALWAYS)
+        self.scripts_dir = paths.get_ipath_cur('scripts')
```

Comment:
You want `get_cpath` for this and the line two lines below this. 
`/var/lib/cloud/scripts/{per-boot,per-instance,per-once}`

There's no boot distinction (or sub-directories) in `/var/lib/cloud/instance/scripts/`. Anything placed there will run every boot.

Reply:
If I'm reading these correctly ...

* It sounds like the INCLUSION_TYPES_MAP is doing doubleduty: it's both a whitelist of mime-types to suppress warnings from `cloud-init devel make-mime`, and a mapping of prefixes to mime-types. So if I include an 'identity mapping' of 'x-shellscript-per-xxx': 'x-shellscript-per-xxx' for all 3 frequencies, I'll benefit from the first feature and vestigially participate in the second.

* `paths.get_ipath_cur()` returns /var/cloud/instance/scripts, and I need /var/cloud/scripts, which can get by just switching to `paths.get_cpath()`.

* You'd like to see a base class and 3 subclasses. It's too bad I can't go one better, and just write a single class, and let `handle_parts()`do the right thing based on frequency. But it looks like I can't, because of the Handler subclass:

```
    def __init__(self, paths, **_kwargs):
        Handler.__init__(self, PER_ONCE)
```

If we don't actually need to call Handler's c'tor, and/or that frequency doesn't actually matter, then we can get away with a single class. But that feels sloppy. If I have to use 3 subclasses that's ok.

Is that everything? What am I missing ...